### PR TITLE
further sample improvements

### PR DESCRIPTION
**Pull request by derisen**
# CHANGELOG

* Updated Angular 8 to Angular 9.
* Added a new route protected by MSALGuard.
* Added Angular Factory for msal configuration objects.
* Added error handling for interceptor errors ([see](https://github.com/Azure-Samples/active-directory-javascript-singlepageapp-angular/pull/5))
* Reorganized configuration file and added further clarifications.
* Minor changes on README.md.

p.s. most of these changes are taken directly from msal-angular dev sample 9

### CODE REVIEW

File: tsconfig.app.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "src/polyfills.ts"
   ],
   "include": [
-    "src/**/*.ts"
+    "src/**/*.d.ts"
```
Comment (jasonnutter) [inline tsconfig.app.json:12]: Why is this changed?

Comment (derisen) [reply tsconfig.app.json]: I honestly can't remember doing this :/ must be unintentional -reverting back..

**Comment by derisen** [comment]
> One question otherwise looks good.

Thanks! Will update msal-angular and add password reset handling in the next PR.
